src/adapters/Cornerstone3D/EllipticalROI.js

```diff
@@ -0,0 +1,167 @@
+import MeasurementReport from "./MeasurementReport";
+import TID300Ellipse from "../../utilities/TID300/Ellipse";
+import CORNERSTONE_3D_TAG from "./cornerstone3DTag";
+
+const ELLIPTICALROI = "EllipticalROI";
+const FINDING = "121071";
+const FINDING_SITE = "G-C0E3";
+
+const trackingIdentifierTextValue = "Cornerstone3DTools@^0.1.0:EllipticalROI";
+
+class EllipticalROI {
+    constructor() {}
+
+    // TODO: this function is required for all Cornerstone Tool Adapters, since it is called by MeasurementReport.
+    static getMeasurementData(MeasurementGroup, imageId, imageToWorldCoords) {
+        const {
+            defaultState,
+            NUMGroup,
+            SCOORDGroup
+        } = MeasurementReport.getSetupMeasurementData(MeasurementGroup);
+
+        const { GraphicData } = SCOORDGroup;
+
+        // GraphicData is ordered as [majorAxisStartX, majorAxisStartY, majorAxisEndX, majorAxisEndY, minorAxisStartX, minorAxisStartY, minorAxisEndX, minorAxisEndY]
+        // But Cornerstone3D points are ordered as top, bottom, left, right for the
```

Comment:
> so we need to identify if the majorAxis is horizontal or vertical

Do we? Won't CS3D just auto correct this when you drag a point, and all stat calculations should be correct anyway iirc?

Reply:
No they are always top, bottom, left, right when I checked